lib/elephant_in_the_room/sites/sites.ex

```diff
@@ -361,6 +361,25 @@ defmodule ElephantInTheRoom.Sites do
     |> Repo.all
   end
 
+  def get_columnists_and_posts(%{id: site_id}, amount) do
+    query = from post in Post,
+      where: post.site_id == ^site_id,
+      order_by: [desc: post.inserted_at],
+      distinct: post.author_id,
+      left_join: author in Author,
```

Comment:
If you use an inner join you'll get only posts that have an author. No need to do the filtering below.

Reply:
Changed